component/src/dex/position_manager.rs

```diff
@@ -41,6 +42,17 @@ pub trait PositionManager: StateWrite + PositionRead {
         position: Position,
         initial_reserves: Reserves,
     ) -> Result<LpNft> {
+        // We limit the sizes of reserve amounts to at most 112 bits. This is to give us extra
+        // headroom to perform intermediary calculations during composition.
+        // TODO: remove the extra casting once `Amount` gets full 128 bits support.
+        if initial_reserves.r1.value() as u128 > MAX_RESERVE_AMOUNT
+            || initial_reserves.r2.value() as u128 > MAX_RESERVE_AMOUNT
```

Comment:
We'll want to check the position coefficients as well as the reserves.

In terms of structuring the checks, ideally we'd check this before starting execution, and then again as a sanity check here.  So I think it'd make sense to factor the check out into methods on `Position` and `Reserves`, maybe a `fn check_bounds(&self) -> Result<()>` on each, so we have the check defined only in one place.

Reply:
thanks, this results in a very legible `PositionManager::position_open`